behave/features/steps/common_steps.py

```diff
@@ -106,9 +112,29 @@ def step_impl(context):
                 context.helperfunc.find_by_name("diagnosis-next").click()
             except StaleElementReferenceException:
                 context.helperfunc.find_by_name("diagnosis-next").click()
+            # This is required because the diagnosis-next button on the current page and next page have the same name
+            # Without this sleep it will just find the same button and click it again instead of waiting for new button
+            # to load
+            time.sleep(1)
```

Comment:
Wondering if it is possible to do this with a wait until rather than a sleep?

Reply:
@jogilder The next button class does not change between screens. So if you wait for the next button it will just return really quickly picking up the current next button instead of waiting for the next one. The sleep will ensure that the new next button has time to load

Reply:
@said-moj, I understand why you added it - am wondering if there is a way you can do this along the lines of wait_until that is used elsewhere rather than an explicit sleep. Is there another change we can wait for?

Reply:
Without a sleep the wait_until will suffer from the same problem, in that it will pass in the first evaluation
i.e
```
       .
       .
       .
       context.helperfunc.find_by_name("diagnosis-next").click()
       .
       .
       .
        def wait_for_next_button(*args):
            return context.helperfunc.find_by_name("diagnosis-next") is not None
       wait = WebDriverWait(context.helperfunc.driver(), 10)
       wait.until(wait_for_next_button)
```
This would return True on the first pass because it will find the current diagnosis-next immediately.

However
```
       .
       .
       .
       context.helperfunc.find_by_name("diagnosis-next").click()
       .
       .
       .
        def wait_for_next_button(*args):
            time.sleep(1)
            return context.helperfunc.find_by_name("diagnosis-next") is not None
        wait = WebDriverWait(context.helperfunc.driver(), 10)
        wait.until(wait_for_next_button)
```
This will work but all it did was wrap the time.sleep in a wait where time.sleep is actually doing the waiting(sleeping)

Reply:
Cool. Just wondered.

Reply:
@psweeting1 is that a comment for me or @jogilder